pkg/authz/warrant/sqlite.go

```diff
@@ -0,0 +1,461 @@
+package authz
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/warrant-dev/warrant/pkg/database"
+	"github.com/warrant-dev/warrant/pkg/middleware"
+	"github.com/warrant-dev/warrant/pkg/service"
+)
+
+type SQLiteRepository struct {
+	database.SQLRepository
+}
+
+func NewSQLiteRepository(db *database.SQLite) SQLiteRepository {
+	return SQLiteRepository{
+		database.NewSQLRepository(&db.SQL),
+	}
+}
+
+func (repo SQLiteRepository) Create(ctx context.Context, warrant Warrant) (int64, error) {
+	var newWarrantId int64
+	now := time.Now().UTC()
+	err := repo.DB.GetContext(
+		ctx,
+		&newWarrantId,
+		`
+			INSERT INTO warrant (
+				objectType,
+				objectId,
+				relation,
+				subjectType,
+				subjectId,
+				subjectRelation,
+				contextHash,
+				createdAt,
+				updatedAt
+			) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)
+			ON CONFLICT (objectType, objectId, relation, subjectType, subjectId, subjectRelation, contextHash) DO UPDATE SET
+				createdAt = ?,
+				deletedAt = NULL
+			RETURNING id
+		`,
+		warrant.ObjectType,
+		warrant.ObjectId,
+		warrant.Relation,
+		warrant.SubjectType,
+		warrant.SubjectId,
+		warrant.SubjectRelation,
+		warrant.ContextHash,
+		now,
+		now,
+		now,
+	)
+	if err != nil {
+		// TODO: Cast to appropriate SQLite error
```

Comment:
We actually don't need this since the query above will handle duplicate record scenarios (by updating the created at). Also, we handle duplicate record scenarios (for non-deleted records) at the service layer by doing a `Get` before the `Create`.